Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SpawnerComponent.h"
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
```

Comment:
Is this include necessary? If not , also remove dependency from cmake

Reply:
The Field in line `serialize->Class<ROS2SpawnerComponent, AZ::Component>()->Version(1)->Field("Spawner", &ROS2SpawnerComponent::m_spawnables);` does not work without this include.

Reply:
The include missing from `AzToolsFramework/Entity/EditorEntityHelpers.h` is `#include <AzToolsFramework/API/EntityCompositionRequestBus.h>` but it does not remove the dependence on AzToolsFramework. I checked the includes deeper but the farthest I can get is `AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI.h` which still does not solve the problem.

The dependency in the CMake is already added by another PR.